File: orchalang-orchacompiler/src/main/java/transactionJPA/StudentDomain.java
package transactionJPA;

import javax.persistence.*;

@Entity(name = "Student")
@Table(name = "Student")
public class StudentDomain {

    public StudentDomain(String firstName, int age) {
        this.firstName = firstName;
        this.age = age;
    }

    public StudentDomain(){
    }

    @Id
    @Column(name = "rollNumber")
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long rollNumber;

    @Column(name = "firstName")
    private String firstName;

    @Column(name = "age")
    private int age;

    public Long getRollNumber() {
        return rollNumber;
    }

    public void setRollNumber(Long rollNumber) {
        this.rollNumber = rollNumber;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    @Override
    public String toString() {
        return "StudentDomain{" +
                "rollNumber=" + rollNumber +
                ", firstName='" + firstName + '\'' +
                ", age=" + age +
                '}';
    }
}

File: orchalang-orchacompiler/src/main/kotlin/orcha/school/StudentDomain.kt
package orcha.school

import javax.persistence.*

@Entity(name = "Student")
@Table(name = "Student")
class StudentDomain {
    constructor(firstName: String?, age: Int) {
        this.firstName = firstName
        this.age = age
    }

    constructor(firstName: String?, age: Int, rollNumber: Long) {
        this.firstName = firstName
        this.age = age
        this.rollNumber = rollNumber
    }

    constructor() {}

    @Id
    @Column(name = "rollNumber")
    @GeneratedValue(strategy = GenerationType.AUTO)
    var rollNumber: Long? = null

    @Column(name = "firstName")
    var firstName: String? = null

    @Column(name = "age")
    var age = 0

    override fun toString(): String {
        return "StudentDomain{" +
                "rollNumber=" + rollNumber +
                ", firstName='" + firstName + '\'' +
                ", age=" + age +
                '}'
    }
}